### ISSUE TITLE
Add a map to display where our contacts are

### ISSUE THREAD

**Issue by djaiss** [monicahq/monica#391]
This could be a discreet map button on the people list, in which we could display a map with dots representing our contacts.

**Comment by turtles2** [comment]
@djaiss How do you propose we get the users locations. I see two ways firstly we could geocode there address but this requires some geocoding API most have limits and restrictions the ones that don't would require us to use our own software and use OpenAddress data which is not complete.  The second approach is we could display a map and have the user drop a pin kinda of a manual geocoding.  I have experience with both approaches.

**Comment by djaiss** [comment]
@turtles2 I think it could be even simpler than this. It's not `user location`, it's `contact location`.
We could simply take the address stored per contact, manually entered by the contact.
Can we then display them on a map using Google Maps for instance?

If we can find the easiest way to achieve this, I'll wireframe where I think we should put the map

**Comment by turtles2** [comment]
If we are displaying the map on the page my concern is not technical it's
legal as Google has strict rules for the use of there APIs and last time I
did something similar anything behind a log in required a payment for the
API. Now if we are talking about linking to Google maps then we would avoid
that but the user would leave the application. The other approach uses open
data and software to power the maps so there is no thorny legal issues.

On Jan 6, 2018 11:10 PM, "Régis Freyd" <notifications@github.com> wrote:

@turtles2 <https://github.com/turtles2> I think it could be even simpler
than this. It's not user location, it's contact location.
We could simply take the address stored per contact, manually entered by
the contact.
Can we then display them on a map using Google Maps for instance?

If we can find the easiest way to achieve this, I'll wireframe where I
think we should put the map

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/monicahq/monica/issues/391#issuecomment-355800161>, or mute
the thread

**Comment by djaiss** [comment]
@turtles2 so OpenAddress then? What is the library for this? Would be MIT?

**Comment by turtles2** [comment]
Here is the thing with Open Address I am only aware of one service offering
an API and they are shutting down. The code that powers it is all open
source (MIT?) so I will check if someone else is going to pick it up
otherwise it will be difficult as it requires allot of storage space to run
we are talking somewhere around 100 to 200 GB of data. The service is
mapzen by the way.

**Comment by djaiss** [comment]
@turtles2 ouch. 200 Gb. We have to find another solution.
I've taken a look at Google Maps API. For personal use, we don't need to provide payment information up to a reasonable amounts of queries per day. For our `.com` version, I'd need to provide payment information though.
I think that's the only viable route for now.

**Comment by turtles2** [comment]
I agree with you on that. Why don't you put in a wire frame and we will see
if we can get this done.

**Comment by rogerxaic** [comment]
Hi there,

How about using [LeafletJS](https://leafletjs.com/) to display [OpenStreetMap](https://www.openstreetmap.org/) maps?
They've also got [Nominatim](https://nominatim.openstreetmap.org), a geocoding tool.

I don't know though if they've got an open API or if a manual clone+deploy is necessary? Or maybe there's an API but an API KEY is required? 

An option could be to have a disabled feature and someone that wants to activate should get an API key and set it up via env vars or the admin panel described in #1876 and #3462.

What do you think?

**Comment by djaiss** [comment]
No, the feature should not be enabled/disabled in the Settings panel.

It should be visible in the app if the key is there in the env file, and should be hidden if the key is not there. But we shouldn't have another option in the Settings panel.

Monica already supports geolocation with Location IQ. We just need to find the proper map provider.

**Comment by virtadpt** [comment]
@djaiss +1 for Open Streetmap.

**Comment by iloveitaly** [comment]
This would be awesome. I think there's already a geocoding function built in, so there shouldn't be a massive lift.
